### ISSUE TITLE
Support for content inside components

### ISSUE THREAD

**Issue by benjamminf** [monkberry/monkberry#6]
Would be interesting if components could support inner content. Vue uses it's `<slot>` tags to accomplish this, which I quite like. Twig also has a similar approach using `{% block %}` tags.

I attempted to add inner content to a component, and while it compiled just fine, there was some strange behaviour. I'm not sure if this is something already implemented but just not documented.

**Comment by elfet** [comment]
Hi,

Now custom tags in Monkberry behave as inline components in vue. I was thinking about adding support for slots and block in future. But i need more use cases, to better understand what API should it have.

**Comment by benjamminf** [comment]
Here's a use case for a project I'm working on:

I have a header bar that contains the title of the page. The settings page requires a Save button to be added to the right of the title. The data graphing page requires Start, Stop and Export buttons. The buttons here are dependent on the page, and not the header bar, so it doesn't make sense to bundle them inside the header bar component. But I still need a way of displaying these buttons _in_ the header bar.

This application was written in Vue, and I used slots to solve this problem. I'm currently experimenting with replacing Vue with Monkberry, and I'm really liking how clean and flexible Monkberry's API is. So far, the only thing I can't seem to elegantly rebuild in Monkberry is this.

Thanks for your hard work, I'm really enjoying using it 👍

**Comment by elfet** [comment]
for now i see two options for this: using block and using slots.

### Block

```twig
<header>
  {{ title }}
  {% block button %}{% endblock %}
</header>
```

```twig
<header title="...">
  {% block button %}
    <button>
  {% endblock %}
</header>
```

### Slot

```twig
<header>
  {{ title }}
  <slot name="button"/>
</header>
```

```twig
<header title="...">
  <slot name="button">
    <button>
  </slot>
</header>
```

With block also possible to implement extending.

**Comment by elfet** [comment]
I like block way, because it allow also to implement template inherence, but need to find way to passing default data to block:

```twig
<Component>
    <div>...</div>
</Component>
```

```twig
<div>
    {% block %}{% endblock %}
</div>
```

**Comment by benjamminf** [comment]
Yeah I'm a fan of the `{% block %}` tag as well. It's a specialised tag, so it should stand out syntactically. Reserving a HTML tag name feels a bit too much like a hack. And as you said, it allows for template inheritance.
